arch/eventloop_posix/eventloop_posix_udp.c

```diff
@@ -997,6 +997,29 @@ UDP_shutdownConnection(UA_ConnectionManager *cm, uintptr_t connectionId) {
     return UA_STATUSCODE_GOOD;
 }
 
+static bool
+#ifdef _WIN32
+isAddressValid(LPSOCKADDR addr) {
+    char sourceAddr[64];
+    if(addr->sa_family == AF_INET) {
+        inet_ntop(AF_INET, &((struct sockaddr_in *)addr)->sin_addr, sourceAddr,
+                  sizeof(sourceAddr));
+#else
+isAddressValid(struct ifaddrs *ifa) {
+    char sourceAddr[NI_MAXHOST];
+    if(ifa->ifa_addr->sa_family == AF_INET) {
+        getnameinfo(ifa->ifa_addr,
+                    (ifa->ifa_addr->sa_family == AF_INET) ? sizeof(struct sockaddr_in)
+                                             : sizeof(struct sockaddr_in6),
+                sourceAddr, NI_MAXHOST, NULL, 0, NI_NUMERICHOST);
+#endif
+        return !(sourceAddr[0] == '1' && sourceAddr[1] == '6' && sourceAddr[2] == '9' &&
```

Comment:
"169.254.0.0/16" is AutoIP and might be a totally valid IP Address. Why do you exclude it?

PS: Binary checks should be always preffered over string comparisons.